keps/sig-node/3317-user-namespaces/README.md

```diff
@@ -0,0 +1,1015 @@
+created to transform the IDs for that pod. The [volumeHost
+interface][volumeHost-interface] will then have this new method:
+
+```
+GetHostIDsForPod(pod *v1.Pod, containerUID, containerGID *int64) (hostUID, hostGID *int64, err error)
+```
+
+This method will only transform the IDs if the pod is running with userns
+enabled. If userns is disabled, the params will not be translated and, also, 
+GetHostIDsForPod() will return nil if the containerUID/containerGID received
+where nil. This is to keep the current functionality untouched, as [fsUser and
+fsGroup can be set to nil][operation-executor] today by the operation executor.
+
+Secondly, we need to modify these volume types to honor the `fsUser` passed in
+mounterArgs to their SetUpAt() method. As the [AtomicWriter already knows and
+honors the FsUser field][atomic-writer-fsUser] already, just setting this field
+is missing when creating the FileProjection (only done when the feature gate is
+enabled). For example, configmaps will just need to set FsUser alongside Data
+and Mode [here][configmap-fsuser] (and all the other paths in that function that
+create the projection, of course). Other volume types are quite similar too.
+
+For this we need to modify the MakePayload() or CollectData() functions of each
+supported volume type to handle the fsUser new param, and the corresponding
+volume SetUpAt() function to pass this new param when calling AtomicWriter.
+
+We have written already this code and the per-volume diff is 3 lines.
+
+For fsGroup() we don't need any changes, these volume types already honors it.
+
+[volume-mounter-args]: https://github.com/kubernetes/kubernetes/blob/cfd69463deeebfc5ae8a0813d7d2b125033c4aca/pkg/volume/volume.go#L125-L129
+[operation-executor]: https://github.com/kubernetes/kubernetes/blob/cfd69463deeebfc5ae8a0813d7d2b125033c4aca/pkg/volume/util/operationexecutor/operation_generator.go#L674-L675
+[volumeHost interface]: https://github.com/kubernetes/kubernetes/blob/cfd69463deeebfc5ae8a0813d7d2b125033c4aca/pkg/volume/plugins.go#L360-L361
+[atomic-writer-fsUser]: https://github.com/kubernetes/kubernetes/blob/cfd69463deeebfc5ae8a0813d7d2b125033c4aca/pkg/volume/util/atomic_writer.go#L68
+[configmap-fsuser]: https://github.com/kubernetes/kubernetes/blob/cfd69463deeebfc5ae8a0813d7d2b125033c4aca/pkg/volume/configmap/configmap.go#L272-L273
+
+#### Phase 2: pods with volumes
+
+This phase makes user namespaces work in pods with volumes too. This is
+activated via the bool `pod.spec.HostUsers` too and pods fall into this mode if
+some other volume type than the ones listed for phase 1 is used. IOW, when phase
+2 is implemented, pods that use volume types not supported in phase 1, fall into
+the phase 2.
+
+All pods in this mode will use _the same_ mapping, chosen by the kubelet, with a
+length 65535, and mapping the range 0-65534 too. IOW, each pod will have its own user
+namespace, but they will map to _the same_ UIDs/GIDs in the host.
```

Comment:
Forgive me my ignorance, but shouldn't the range in this case (sharing multiple pods) be larger?